lib/shopify_api/resources/fulfillment_order.rb

```diff
@@ -12,5 +12,41 @@ def fulfillments(options = {})
       fo_fulfillments = get(:fulfillments, options)
       fo_fulfillments.map { |fof| FulfillmentOrderFulfillment.new(fof.as_json) }
     end
+
+    def move(new_location_id:)
+      body = { fulfillment_order: { new_location_id: new_location_id } }.to_json
+      keyed_fos = load_keyed_attributes_from_response(post(:move, {}, body))
+      if keyed_fos&.fetch('original_fulfillment_order', nil)&.attributes
+        load(keyed_fos['original_fulfillment_order'].attributes, false, true)
+      end
+      keyed_fos
+    end
+
+    def cancel
+      keyed_fos = load_keyed_attributes_from_response(post(:cancel, {}, only_id))
+      if keyed_fos&.fetch('fulfillment_order', nil)&.attributes
+        load(keyed_fos['fulfillment_order'].attributes, false, true)
+      end
+      keyed_fos
+    end
+
+    def close
+      load_attributes_from_response(post(:close, {}, only_id))
+    end
+
+    private
+
+    def load_keyed_attributes_from_response(response)
+      return load_attributes_from_response(response) if response.code != '200'
```

Comment:
could you please expand on what the structure returned in this case would look like ?

If I'm following things correctly, we will return here if the call either returned a `404` or a `422` with an error message, so are you just loading that error message ?

Reply:
The idea here is that `load_keyed_attributes_from_response` reads multiple items in the response when OK (200) otherwise leverages the error handling in the commonly used `load_attributes_from_response`, which I'm guessing doesn't actually load anything just passes the error code / message back through in some way.